### PR TITLE
Skincare quiz

### PR DESCRIPTION
**Pull request by hernandezvictoria**
## Description

- added initial skincare quiz
- UI handles input data, backend handles adding it to user profile
- also added database on backend that stores product info
- added 10 products with ingredients, looking to add more

## Milestones
- finished user input of skin types and concerns

## Resources
- tutorial on how to do the multiple checkboxes form

## Test Plan
- edge case where user submits form without info, causing error message, then re-inputs data, then resubmits form (this was causing a bug originally)
demo:
https://www.loom.com/share/8d666f08e9aa420dadb3b3e9331022ae?sid=0b936077-d4b6-4a44-a67e-e60eb76dfa11

### CODE REVIEW

File: frontend/meta-capstone/src/components/Quiz.jsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+        setFormSubmitted(true);
+    }
+
+    useEffect(() => {
+        if(formSubmitted && message.type !== "error"){
+            navigate("/home");
+        }
+
+    }, [formSubmitted, message]);
```
[the remark this answers is not in the document]
Comment (hernandezvictoria) [reply frontend/meta-capstone/src/components/Quiz.jsx:98]: I originally did not have the form submitted use effect, and was navigating to home directly, but this caused a bug where if the user wrongly input the form (i.e. didn't select on of each of skin concerns and skin types), they would have to submit twice for the form to submit.

File: backend/routes/products.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+router.get('/products', async (req, res) => {
+
+    if (!req.session.userId) {
+        return res.status(401).json({ error: "you must be logged in to perform this action" })
+    }
+
+
+
+    try {
+        const allProducts = await prisma.productInfo.findMany();
```
[the remark this answers is not in the document]
Comment (hernandezvictoria) [reply backend/routes/products.js:17]: yes! will try to handle that today or tomorrow!

File: backend/prisma/schema.prisma [hunk 3]
```diff
@@ -18,8 +12,8 @@ model User{
   id Int @id @default(autoincrement())
   username String @unique
   password String
-  concerns String?
-  skin_type String?
+  concerns Json?
```
[the remark this answers is not in the document]
Comment (hernandezvictoria) [reply backend/prisma/schema.prisma:15]: oo will look into this

File: backend/routes/products.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+const prisma = new PrismaClient()
+const router = express.Router()
+
+// http://localhost:3000/products
+router.get('/products', async (req, res) => {
+
+    if (!req.session.userId) {
+        return res.status(401).json({ error: "you must be logged in to perform this action" })
+    }
+
```
[the remark this answers is not in the document]
Comment (hernandezvictoria) [reply backend/routes/products.js:13]: handled

File: frontend/meta-capstone/src/components/home_components/ProductContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+function ProductContainer({data, setData}) {
+
+    const fetchAllData = async () => {
+
+      fetch("http://localhost:3000/products", { credentials: "include" })
+        .then((response) => response.json())
+        .then((res) => { setData(res)})
+        .catch(() => {
+          console.error("unable to fetch products: ", error);
```
[the remark this answers is not in the document]
Comment (hernandezvictoria) [reply frontend/meta-capstone/src/components/home_components/ProductContainer.jsx:15]: HANDLED

File: backend/enums.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const SkinTypes = Object.freeze({
+    OILY: "oily",
+    DRY: "dry",
+    SENSITIVE: "sensitive",
+    COMBINATION: "combination",
+    NORMAL: "normal"
+});
+
+const SkinConcerns = Object.freeze({
```
[the remark this answers is not in the document]
Comment (hernandezvictoria) [reply backend/enums.js:9]: ended up inputting the skin concerns manually since im already building the DB by hand, so no longer need to compare the ingredients to the ingredients dataset to get concerns; but still wanna use the ingredients dataset for highlighting key ingredients

File: frontend/meta-capstone/src/components/Quiz.jsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+        // collect selected concerns
+        form.querySelectorAll('input[name="concern"]:checked').forEach((input) => {
+            selectedConcerns.push(input.value);
+        });
+
+        if (selectedSkinTypes.length === 0 || selectedConcerns.length === 0) {
+            setMessage({ type: "error", text: "you must select at least one of each" });
+        } else {
+            setMessage({ type: "success", text: "form submitted successfully!" });
```
[the remark this answers is not in the document]
Comment (hernandezvictoria) [reply frontend/meta-capstone/src/components/Quiz.jsx:35]: handled

File: frontend/meta-capstone/src/components/Quiz.jsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+    }, [formSubmitted, message]);
+
+    return (
+        <>
+            <h2>tell us about your skinterests, {user.username}</h2>
+
+            <form onSubmit={handleSubmit}>
+                <h3>skin type(s) </h3>
+                <input type="checkbox" name="type" id={SkinTypes.OILY} value={SkinTypes.OILY}/>
```
[the remark this answers is not in the document]
Comment (hernandezvictoria) [reply frontend/meta-capstone/src/components/Quiz.jsx:106]: handled
